util/git/client.go

```diff
@@ -223,26 +222,10 @@ func GetRepoHTTPClient(repoURL string, insecure bool, creds Creds, proxyURL stri
 			return customHTTPClient
 		} else if len(serverCertificatePem) > 0 {
 			certPool := certutil.GetCertPoolFromPEMData(serverCertificatePem)
-			customHTTPClient.Transport = &http.Transport{
-				Proxy: proxyFunc,
-				TLSClientConfig: &tls.Config{
-					RootCAs:              certPool,
-					GetClientCertificate: clientCertFunc,
-				},
-				DisableKeepAlives: true,
-			}
-		} else {
-			// else no custom certificate stored.
-			customHTTPClient.Transport = &http.Transport{
-				Proxy: proxyFunc,
-				TLSClientConfig: &tls.Config{
-					GetClientCertificate: clientCertFunc,
-				},
-				DisableKeepAlives: true,
-			}
+			transport.TLSClientConfig.RootCAs = certPool
```

Comment:
Not a requirement, but if you had a minute to update `TestCustomHTTPClient` to cover this block, that would be awesome.  :-)